metric/system/process/helpers_others.go

```diff
@@ -0,0 +1,36 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build !windows
+
+package process
+
+import (
+	"errors"
+	"syscall"
+)
+
+func CanDegrade(err error) bool {
```

Comment:
I like having errors categorized as non-fatal in this package, but I don't like pushing the logic of what errors should result in particular agent statuses here. This package should not be aware of things like setting a process to the degraded state.

Another approach might be to explicitly define a NonFatalError type that you can always return, and then the calling code can check for whether a particular error is NonFatal.

There are a couple of ways you could do this as examples linked in https://go.dev/blog/error-handling-and-go. For grouping all these errors into one NonFatalError type, see something like https://pkg.go.dev/encoding/json#SyntaxError. 

The net package also has a few types with an https://pkg.go.dev/net#AddrError.Temporary method, an IsNonFatal might work but I think this might be a bit more work to maintain.

Reply:
It actually looks like we already have a `NonFatalErr` type in here specifically for this, it probably just isn't hooked up everywhere it should be:

https://github.com/elastic/elastic-agent-system-metrics/blob/a9ecc37f3de53d18a85ad182718010650aad774b/metric/system/process/process.go#L222-L233

Reply:
I've hooked up `NonFatalError`. Callers will do the needful and check for errors.